data-tck/src/main/java/io/micronaut/data/tck/entities/ShipmentWithIndexOnFieldsCompositeIndexes.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.tck.entities;
+
+import io.micronaut.core.annotation.Creator;
+import io.micronaut.data.annotation.GeneratedValue;
+import io.micronaut.data.annotation.Index;
+import io.micronaut.data.annotation.MappedEntity;
+
+import javax.persistence.Column;
+import javax.persistence.Id;
+import java.util.Objects;
+
+@MappedEntity
+public class ShipmentWithIndexOnFieldsCompositeIndexes {
+
+    @Creator
+    public ShipmentWithIndexOnFieldsCompositeIndexes(Long shipmentId, String field, String taxCode) {
+        this.shipmentId = shipmentId;
+        this.field = field;
+        this.taxCode = taxCode;
+    }
+
+    // for hibernate
+    public ShipmentWithIndexOnFieldsCompositeIndexes() {
+    }
+
+    @Id
+    @GeneratedValue
+    private Long shipmentId;
+
+    @Column(name = "field")
+    @Index(columnList = {"field, taxCode"}, unique = true)
```

Comment:
I think should be an array now

Reply:
I think list here is used more in the sense of "a sequence of elements" rather than referring to a specific data structure.

Reply:
@dstepanov  Maybe _columnNames_ or just _columns_ are a better match?

Reply:
Yeah ‘columns’ should be fine

Reply:
@dstepanov done

- [x]